examples/autoignition-mpi.py

```diff
@@ -499,25 +500,75 @@ def my_get_timestep(t, dt, state):
                                 op="max")
         return ts_field, cfl, min(t_remaining, dt)
 
-    def limiter(cv, temp=None):
+    def limiter(cv, pressure, temperature):
         spec_lim = make_obj_array([
-            bound_preserving_limiter(dcoll, cv.species_mass_fractions[i], mmax=1.0)
+            bound_preserving_limiter(dcoll, cv.species_mass_fractions[i], mmin=0.0)
             for i in range(nspecies)
         ])
 
-        kin_energy = 0.5*np.dot(cv.velocity, cv.velocity)
+        aux = cv.mass*0.0
+        for i in range(0, nspecies):
+            aux = aux + spec_lim[i]
+        spec_lim = spec_lim/aux
+
+        mass_lim = eos.get_density(pressure=pressure,
+            temperature=temperature, species_mass_fractions=spec_lim)
 
-        energy_lim = cv.mass*(
-            gas_model.eos.get_internal_energy(temp, species_mass_fractions=spec_lim)
-            + kin_energy
+        energy_lim = mass_lim*(gas_model.eos.get_internal_energy(
+            temperature, species_mass_fractions=spec_lim)
+            + 0.5*np.dot(cv.velocity, cv.velocity)
         )
 
-        return make_conserved(dim=dim, mass=cv.mass, energy=energy_lim,
-                       momentum=cv.momentum, species_mass=cv.mass*spec_lim)
+        cv_limited = make_conserved(dim=dim,
+                                    mass=mass_lim,
+                                    energy=energy_lim,
+                                    momentum=mass_lim*cv.velocity,
+                                    species_mass=mass_lim*spec_lim)
+
+        return cv_limited
+
+    apply_limiter = actx.compile(limiter)
+
+    def _update_dv(cv, temperature):
+        return update_dependent_vars(cv, temperature, eos)
+
+    update_dv = actx.compile(_update_dv)
+
+    def _update_tv(cv, dv):
+        return gas_model.transport.transport_vars(cv, dv, eos)
+
+    update_tv = actx.compile(_update_tv)
+
+    def _update_fluid_state(cv, dv, tv=None):
+        if viscous_terms_on:
+            return ViscousFluidState(cv, dv, tv)
+        else:
+            return FluidState(cv, dv)
+
+    update_fluid_state = actx.compile(_update_fluid_state)
 
     def my_pre_step(step, t, dt, state):
         cv, tseed = state
-        fluid_state = construct_fluid_state(limiter(cv, temp=tseed), tseed)
+
+        # update temperature value
+        fluid_state = construct_fluid_state(cv, tseed)
+
+        # apply limiter and reevaluate CV
+        limited_cv = apply_limiter(
+            fluid_state.cv, fluid_state.pressure, fluid_state.temperature)
+
+        # since temperature is not changed during the limiting process, it is
+        # possible to avoid recomputing it by NOT calling make_fluid_state.
+        # Thus, evaluate ONLY other DV and TV variables and reassemble the state
+        new_dv = update_dv(limited_cv, fluid_state.temperature)
```

Comment:
@MTCam will be able to provide more specific guidance on this, but I wonder if it would be better to modify `eos.dependent_vars` to take in an optional `temperature` argument instead of adding a separate `update_dependent_vars` function.